src/Symfony/Component/Validator/Tests/Constraints/AbstractConstraintValidatorTest.php

```diff
@@ -53,6 +53,10 @@
 
     protected function setUp()
     {
+        if (Validation::API_VERSION_2_5 !== $this->getApiVersion()) {
```

Comment:
Technically, tests for the 2.5-BC API layer should not be allowed to trigger the warnings (the code should still rely on the 2.5 API when it has access to both APIs)